cmd/cluster/cluster.go

```diff
@@ -27,6 +27,7 @@ func NewCreateCommands() *cobra.Command {
 
 	cmd.MarkFlagsMutuallyExclusive("service-cidr", "default-dual")
 	cmd.MarkFlagsMutuallyExclusive("cluster-cidr", "default-dual")
+	cmd.MarkFlagsMutuallyExclusive("machine-cidr", "default-dual")
```

Comment:
Please add these to the validation func on the `core.DefaultOptions()` - we have users of cluster creation outside of CLI and need to validate the actual options struct, not one implementation of how the data is passed (flags).

Reply:
ack, I'll look at this asap.

Reply:
In here? https://github.com/shiftstack/hypershift/blob/b68f5a62a23ea292c7363ad5413ce1f3d43034b0/cmd/cluster/core/create.go#L541-L542
I think I need a little bit of guidance here on what you expect the test to be, I'll be happy to write it.
Thanks